xs/src/admesh/util.c

```diff
@@ -185,24 +185,74 @@ void calculate_normals(stl_file *stl) {
   }
 }
 
-void stl_transform(stl_file *stl, float *trafo3x4) {
+void stl_transform(stl_file *stl, double const *trafo3x4) {
   int i_face, i_vertex, i, j;
   if (stl->error)
     return;
   for (i_face = 0; i_face < stl->stats.number_of_facets; ++ i_face) {
     stl_vertex *vertices = stl->facet_start[i_face].vertex;
     for (i_vertex = 0; i_vertex < 3; ++ i_vertex) {
       stl_vertex* v_dst = &vertices[i_vertex];
-      stl_vertex  v_src = *v_dst;
-      v_dst->x = trafo3x4[0] * v_src.x + trafo3x4[1] * v_src.y + trafo3x4[2]  * v_src.z + trafo3x4[3];
-      v_dst->y = trafo3x4[4] * v_src.x + trafo3x4[5] * v_src.y + trafo3x4[6]  * v_src.z + trafo3x4[7];
-      v_dst->z = trafo3x4[8] * v_src.x + trafo3x4[9] * v_src.y + trafo3x4[10] * v_src.z + trafo3x4[11];
+      double v_src_x = (double)(v_dst->x);
```

Comment:
Was there a reason you needed to modify admesh? This is technically code external to the Slic3r project and proposed changes should also be pushed upstream.

Reply:
as this is now the remaining function that is called in admesh, I wanted it it to be efficient wrt the checks done on each transform (size, volume, faces) and to have and option to directly assign a transformed mesh. I'll look into purposing the corresponding upstream change.